applications/ConvectionDiffusionApplication/custom_elements/symbolic_d_convection_diffusion_explicit.cpp

```diff
@@ -0,0 +1,1783 @@
+    // Add the residual contribution
+    // Note that the reaction is indeed the formulation residual
+    for (unsigned int i_node = 0; i_node < local_size; i_node++) {
+        #pragma omp atomic
+        r_geometry[i_node].FastGetSolutionStepValue(r_process_info[CONVECTION_DIFFUSION_SETTINGS]->GetReactionVariable()) += rhs[i_node];
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::Initialize(
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    BaseType::Initialize(rCurrentProcessInfo);
+    // Resize and intialize output
+    if (mUnknownSubScale.size() != TNumNodes) // number integration points = number nodes
+        mUnknownSubScale.resize(TNumNodes, false);
+    mUnknownSubScale = ZeroVector(TNumNodes);
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::FinalizeSolutionStep(
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    BaseType::FinalizeSolutionStep(rCurrentProcessInfo);
+    // Element variables
+    ElementVariables rVariables;
+    this->InitializeEulerianElement(rVariables,rCurrentProcessInfo);
+    // Reading integration points and local gradients
+    const auto& r_geometry = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints(this->GetIntegrationMethod());
+
+    // Iterate over integration points to update subscales on the gauss point
+    for (unsigned int g = 0; g < integration_points.size(); g++) {
+        // Caluclate N on the gauss point "g"
+        rVariables.N = row(rVariables.N_gausspoint,g);
+        // Compute tau
+        this->CalculateTau(rVariables);
+        // Retrieve unknown belonging to subgrid scale space on gauss integration point g
+        rVariables.unknown_subscale = mUnknownSubScale(g);
+        // Update unknown belonging to subgrid scale space on gauss integration point g
+        this->UpdateUnknownSubgridScaleGaussPoint(rVariables,g);
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template< unsigned int TDim, unsigned int TNumNodes >
+void SymbolicDConvectionDiffusionExplicit<TDim,TNumNodes>::Calculate(
+    const Variable<double>& rVariable,
+    double& Output,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
```

Comment:
You should check that the `rVariable` argument is the one you expect with an if/else statement. If the if condition is not satisfied, you must call the base class calculate method.

Reply:
Why should I be calling the BaseClass method (which for me is the QS element, and does exactly the same)? And not throwing an error?

Reply:
I have added the if-else, where the else throws an error. Once you comment on this point, I will modify and call the base class. I will push "my" change for now in order to push also all other suggestions you made.

Reply:
No. The method is intended to call the base class. Otherwise you would need to implement all the variables you may want to compute in all the elements. Is the base element the one that throws the error.

Reply:
Ok, done.